controllers/core/openstackversion_controller.go

```diff
@@ -216,17 +208,23 @@ func (r *OpenStackVersionReconciler) Reconcile(ctx context.Context, req ctrl.Req
 		return ctrl.Result{}, err
 	}
 
-	// greenfield deployment //FIXME check dataplane here too
-	if controlPlane.Status.DeployedVersion == nil {
-		Log.Info("No controlplane or controlplane is not deployed")
-		return ctrl.Result{}, nil
+	// lookup nodesets
+	_, dataplaneNodesets, err := openstack.GetDataplaneNodesets(ctx, controlPlane, versionHelper)
+	if err != nil {
+		Log.Error(err, "Failed to get dataplane nodesets")
+		return ctrl.Result{}, err
 	}
 
-	// TODO minor update for OVN Dataplane in progress
+	// greenfield deployment
+	if controlPlane.Status.DeployedVersion == nil && !openstack.DataplaneNodesetsDeployedVersionIsSet(dataplaneNodesets) {
+		Log.Info("Waiting for controlplane and dataplane nodesets to be deployed.")
+		return ctrl.Result{}, nil
+	}
 
 	// minor update for OVN Controlplane in progress
```

Comment:
nit: I think this comment more belongs to the next if now at L227, or need to be reworded that "minore update in progress" as the current condition only checks that.

Reply:
got it fixed